client/src/pages/Recommend/Recommend.tsx

```diff
@@ -0,0 +1,66 @@
+import { useSearchParams } from 'react-router-dom';
+import React, { FC, useEffect, useState } from 'react';
+import { Box, Grid, Paper } from '@mui/material';
+import { useQuery } from '@apollo/client';
+import LinearProgress from '@mui/material/LinearProgress';
+import { Movie } from '../../components/typedefs/typedefs';
+import { MOVIES_BY_IDS_QUERY } from './queries';
+import { MovieCard } from '../../components/MovieCard/MovieCard';
+import { useMovie } from '../../hooks/useMovie';
+
+interface Params {
+  ids: number[],
+  title: string | null,
+}
+
+export const Recommend: FC = () => {
+  const [searchParams] = useSearchParams();
+  const [params, setParams] = useState<Params>({
+    ids: [],
+    title: '',
+  });
+
+  const { selectMovie } = useMovie();
+
+  const {
+    loading,
+    data,
+  } = useQuery(MOVIES_BY_IDS_QUERY, { variables: { ids: params.ids } });
+
+  useEffect(() => {
+    const ids = searchParams.get('ids');
+    const title = searchParams.get('title');
+
+    setParams({
+      ids: ids ? ids.split(',').map((id) => Number(id)) : [],
```

Comment:
!![2342] === !![] - always true
![234] === ![] always true
Use length or just remove
```suggestion
      ids: ids.split(',').map((id) => Number(id)),
```